bgpd/bgp_vty.c

```diff
@@ -7328,7 +7328,7 @@ static int bgp_clear_prefix(struct vty *vty, const char *view_name,
 /* one clear bgp command to rule them all */
 DEFUN (clear_ip_bgp_all,
        clear_ip_bgp_all_cmd,
-       "clear [ip] bgp [<view|vrf> VIEWVRFNAME] [<ipv4|ipv6|l2vpn> [<unicast|multicast|vpn|labeled-unicast|flowspec|evpn>]] <*|A.B.C.D|X:X::X:X|WORD|(1-4294967295)|external|peer-group PGNAME> [<soft [<in|out>]|in [prefix-filter]|out>]",
+       "clear [ip] bgp [<view|vrf> VIEWVRFNAME] [<ipv4|ipv6|l2vpn> [<unicast|multicast|vpn|labeled-unicast|flowspec|evpn>]] neighbor <*|A.B.C.D|X:X::X:X|WORD|(1-4294967295)|external|peer-group PGNAME> [<soft [<in|out>]|in [prefix-filter]|out>]",
```

Comment:
You don't need to add the `neighbor` keyword to get the autocompletion to work.  This is a bit non-obvious, but the autocompletion has 2 alternative ways to trigger itself:

- placeholder name of an input token, e.g. `VIEWVRFNAME` - `WORD` and `NAME` are excluded and assumed to be "no name"
- variable name of an input token

The variable name of an input token is either set manually by using `WORD$varname` or `A.B.C.D$varname` or `<A.B.C.D|X:X::X:X|WORD>$varname` (= sets it for the last token in each of the alternative choices), or is automatically set by the previous non-variable token (i.e. `neighbor WORD` has the same effect as `neighbor WORD$neighbor`.)

bgpd has these autocompletions:

```
bgpd $ git grep -A 1 -B 1 completions 
bgp_vty.c-static const struct cmd_variable_handler bgp_var_neighbor[] = {
bgp_vty.c:      {.varname = "neighbor", .completions = bgp_ac_neighbor},
bgp_vty.c:      {.varname = "neighbors", .completions = bgp_ac_neighbor},
bgp_vty.c:      {.varname = "peer", .completions = bgp_ac_neighbor},
bgp_vty.c:      {.completions = NULL}};
bgp_vty.c-
--
bgp_vty.c-static const struct cmd_variable_handler bgp_var_peergroup[] = {
bgp_vty.c:      {.tokenname = "PGNAME", .completions = bgp_ac_peergroup},
bgp_vty.c:      {.completions = NULL} };
bgp_vty.c-
--
bgpd.c-static const struct cmd_variable_handler bgp_viewvrf_var_handlers[] = {
bgpd.c: {.tokenname = "VIEWVRFNAME", .completions = bgp_viewvrf_autocomplete},
bgpd.c: {.varname = "instasn", .completions = bgp_instasn_autocomplete},
bgpd.c: {.completions = NULL},
bgpd.c-};
```

So... you can change the `clear` command to use `<*|A.B.C.D$neighbor|X:X::X:X$neighbor|WORD$neighbor|(1-4294967295)|external|peer-group PGNAME>`

(Having the same varname more than once is not a problem.)

Reply:
(It's probably enough to use `WORD$neighbor`, I don't 100% remember what happens if you start type `1.2<Tab>`, i.e. it would match on either `A.B.C.D` or `WORD`)

Reply:
@eqvinox awesome! :)